src/homework5/App2.java

```diff
@@ -0,0 +1,62 @@
+package homework5;
+
+import java.util.Scanner;
+
+public class App2 {
+
+    static Scanner scanner;
+    private static double result;
+
+    public static void main(String[] args) {
+        getData();
+        showResult();
+    }
+
+    private static void getData() {
+        scanner = new Scanner(System.in);
+        System.out.println("Enter first number: ");
+        double firstNum = scanner.nextDouble();
+        System.out.println("Enter second number: ");
+        double secondNum = scanner.nextDouble();
+        System.out.println("""
+                Select operation:
+                1 - Sum
+                2 - Difference
+                3 - Multiplication
+                4 - Division
+                0 - Close application
+                """);
+        int selectOperation = scanner.nextInt();
+        calcData(selectOperation, firstNum, secondNum);
+    }
+
+    private static void calcData(int operation, double num1, double num2) {
+        switch (operation) {
+            case 1 -> result = num1 + num2;
+            case 2 -> result = num1 - num2;
+            case 3 -> result = num1 * num2;
+            case 4 -> {
+                result = num1 / num2;
+                if (num2 == 0) {
+                    System.out.println("Can't divide by zero! Try again.");
+                    getData();
```

Comment:
recursive call, not a good idea

Reply:
Replaced recursive call with calcAgain method.